src/Raven.Server/Monitoring/Snmp/SnmpDatabase.cs

```diff
@@ -0,0 +1,212 @@
+using System;
+using System.Collections.Concurrent;
+using System.Threading.Tasks;
+using Lextm.SharpSnmpLib.Pipeline;
+using Raven.Client.Documents.Changes;
+using Raven.Server.Documents;
+using Raven.Server.Monitoring.Snmp.Objects.Database;
+using Raven.Server.Monitoring.Snmp.Objects.Server;
+
+namespace Raven.Server.Monitoring.Snmp
+{
+    public class SnmpDatabase
+    {
+        private readonly ConcurrentDictionary<string, int> _loadedIndexes = new ConcurrentDictionary<string, int>(StringComparer.OrdinalIgnoreCase);
+
+        private readonly DatabasesLandlord _databaseLandlord;
+
+        private readonly ObjectStore _objectStore;
+
+        private readonly string _databaseName;
+
+        private readonly int _databaseIndex;
+
+        private readonly object _locker = new object();
+
+        private bool _attached;
+
+        public SnmpDatabase(DatabasesLandlord databaseLandlord, ObjectStore objectStore, string databaseName, int databaseIndex)
+        {
+            _databaseLandlord = databaseLandlord;
+            _objectStore = objectStore;
+            _databaseName = databaseName;
+            _databaseIndex = databaseIndex;
+
+            Initialize();
+
+            databaseLandlord.OnDatabaseLoaded += loadedDatabaseName =>
```

Comment:
This is per database, but attaches to the global object, need to check that we don't have a leak here again

Reply:
I will verify that, but I'm using the same technique as we used in 3.5, so it should be good.

Reply:
I think this is fine (it is stored always, but there is just a single instance of it per server).

Reply:
Doesn't handle what happens when the db is deleted, though